src/main/java/com/iexec/worker/tee/scone/LasService.java

```diff
@@ -0,0 +1,95 @@
+import com.iexec.worker.docker.DockerService;
+import com.iexec.worker.sgx.SgxService;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class LasService {
+    @Getter
+    private final String containerName;
+    private final String imageUri;
+    @Getter
+    private final SconeConfiguration sconeConfig;
+
+    private final WorkerConfigurationService workerConfigService;
+    private final SgxService sgxService;
+    private final DockerService dockerService;
+
+    @Getter
+    private boolean isStarted;
+
+    public LasService(String containerName,
+                      String imageUri,
+                      SconeConfiguration sconeConfig,
+                      WorkerConfigurationService workerConfigService,
+                      SgxService sgxService,
+                      DockerService dockerService) {
+        this.containerName = containerName;
+        this.imageUri = imageUri;
+        this.sconeConfig = sconeConfig;
+        this.workerConfigService = workerConfigService;
+        this.sgxService = sgxService;
+        this.dockerService = dockerService;
+    }
+
+    boolean start() {
+        if (isStarted) {
+            return true;
+        }
+
+        DockerRunRequest dockerRunRequest = DockerRunRequest.builder()
+                .containerName(containerName)
+                .imageUri(imageUri)
+                // pre-compute, application & post-compute enclaves will be
+                // able to talk to the LAS via this network
+                .dockerNetwork(workerConfigService.getDockerNetworkName())
+                .sgxDriverMode(sgxService.getSgxDriverMode())
+                .maxExecutionTime(0)
+                .build();
+        if (!imageUri.contains(sconeConfig.getRegistryName())) {
+            throw new RuntimeException(String.format("LAS image (%s) is not " +
+                    "from a known registry (%s)", imageUri, sconeConfig.getRegistryName()));
+        }
+        DockerClientInstance client;
+        try {
+            client = dockerService.getClient(
+                    sconeConfig.getRegistryName(),
+                    sconeConfig.getRegistryUsername(),
+                    sconeConfig.getRegistryPassword());
+        } catch (Exception e) {
+            log.error("", e);
+            throw new RuntimeException("Failed to get Docker authenticated client to run LAS");
+        }
+        if (client == null) {
+            log.error("Docker client with credentials is required to enable TEE support");
+            return false;
+        }
+        if (!client.pullImage(imageUri)) {
+            log.error("Failed to download LAS image");
+            return false;
+        }
+
+        DockerRunResponse dockerRunResponse = dockerService.run(dockerRunRequest);
+        if (!dockerRunResponse.isSuccessful()) {
+            log.error("Failed to start LAS service");
+            return false;
+        }
+        return true;
+    }
+
+    void stop() {
```

Comment:
I guess we can go back to `false` when stopping the LAS. In theory, it should never be stopped until the end of execution of the worker so no need for it to be toggled. But let's be clean!
https://github.com/iExecBlockchainComputing/iexec-worker/pull/464/commits/ec44ca0df3596f6a639535499cd32cd46b827397.

Reply:
hmm, sadly, it's not that clean due to `DockerClientInstance` implementation in iexec-common.
stopAndRemoveContainer implementation can produce weird results because their implementation describes what happens instead of the targeted result.
To give an example, they will return `false` if the provided container has already been removed (for any reason).
This will create a lasStarted value of `true`.
We may need to add a TODO to note the current situation is not good and needs to be reviewed cleanly.

Reply:
I have tried to resolve the issue. Please see https://github.com/iExecBlockchainComputing/iexec-worker/pull/464/commits/072fecad7d7aa95da56e19e74e4140611b0c83a3.
The idea is to not depend anymore on `DockerClientInstance::stopAndRemoveContainer` and simply check whether the container is still present after the completion of the method.